entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -89,14 +91,29 @@ public void execute(EntityQueryRequest request, StreamObserver<ResultSetChunk> r
         .convertToEDSQuery(request, attrNameToEDSAttrMap.get(request.getEntityType()));
     Iterator<Document> documentIterator = entitiesCollection.search(
         DocStoreConverter.transform(tenantId.get(), query));
-    List<Entity> entities = convertDocsToEntities(documentIterator);
-
-    //Build result
-    //TODO : chunk response. For now sending everything in one chunk
-    responseObserver.onNext(convertEntitiesToResultSetChunk(
-        entities,
-        request.getSelectionList(),
-        attrNameToEDSAttrMap.get(request.getEntityType())));
+
+    boolean isFirstEntity = true;
+    while (documentIterator.hasNext()) {
+      Optional<Entity> entity = DOCUMENT_PARSER.parseOrLog(documentIterator.next(), Entity.newBuilder());
+      if (entity.isPresent()) {
+        Row row = convertToEntityQueryResult(entity.get(), request.getSelectionList(), attrNameToEDSAttrMap.get(request.getEntityType()));
+        ResultSetChunk.Builder resultBuilder = ResultSetChunk.newBuilder();
+        // Set metadata for first entity
```

Comment:
was the metadata not being sent earlier?

Reply:
It was being sent. Currently, we were sending only one single chunk for all serarched entities.